### PR TITLE
[DOC] Update documentation for EmberArray.find

### PR DESCRIPTION
**Pull request by brunoocasali**
Add code examples for `EmberArray.find` method.

Issue #18228

### CODE REVIEW

**Review by locks — APPROVED** [review]
Appreciate the help! 😁 I'll merge as soon as Browserstack gives us a green light.

**Comment by brunoocasali** [comment]
@locks yeah, I couldn't "retry" the PR but ready to go now!
Next week I will send some more contributions :)

**Comment by locks** [comment]
Keep them coming!

File: packages/@ember/-internals/runtime/lib/mixins/array.js [hunk 1]
```diff
@@ -778,6 +778,20 @@ const ArrayMixin = Mixin.create(Enumerable, {
...
+    ```javascript
+    let users = [
+      { id: 1, name: 'Yehuda' },
+      { id: 2, name: 'Tom' },
+      { id: 3, name: 'Melanie' },
+      { id: 4, name: 'Leah' }
+    ];
+
+    users.find((user) => user.name == 'Tom'); // [{ id: 2, name: 'Tom' }]
+    users.find(({ id }) => id == 3); // [{ id: 3, name: 'Melanie' }]
```
Comment (btecu) [inline packages/@ember/-internals/runtime/lib/mixins/array.js]: Should use `===`

**Pull request merged**